### DISCUSSION TITLE
Round trips with geopandas

### DISCUSSION THREAD

**Discussion by scottyhq** [stac-utils/stac-geoparquet#83] [General]
This library provides really nice functionality for using STAC API responses, and converting to arrow/parquet. 

I was hoping to instead start with a previously saved parquet file to do some filtering with geopandas, but I'm finding that using geopandas.read_parquet doesn't have the expected STAC column structure:

```python
import geopandas as gpd 
import stac_geoparquet 

gf = gpd.read_parquet('tests/data/naip.parquet')
# Optionally do some filtering here, and go back to Pystac.Items / python dicts to use with other tools
rbr = stac_geoparquet.arrow.stac_table_to_items(gf.to_arrow())
list(rbr)  # AssertionError: assert pa.types.is_struct(bbox_col.type)
```

Based on these docs https://stac-utils.github.io/stac-geoparquet/latest/usage/#parquet I also thought to try the following:
```python
table = pyarrow.parquet.read_table('./tests/data/naip.parquet')
gf = gpd.GeoDataFrame.from_arrow(table) #ValueError: No geometry column found in the Arrow table.
```

Is it possible to just use gpd.read_parquet with some kwargs to enable this type of workflow?

**Comment by kylebarron** [comment]
Can you inspect the schema in `gf.to_arrow()`?

**Comment by scottyhq** [comment]
So I used a bad example here and just assumed the test file was a recent geoparquet version `gpq describe tests/data/naip.parquet` -> `│ GeoParquet Version │ 0.4.0`. Doing a round trip starting with valid STAC JSON does work as expected!

```python
# NOTE: this changes STAC bbox lists to dicts: {'xmin': -65.75386, 'ymin': 18.183872, 'xmax': -65.683663, 'ymax': 18.253643}
rbr = stac_geoparquet.arrow.parse_stac_ndjson_to_arrow('tests/data/naip-pc.json')

gf = gpd.GeoDataFrame.from_arrow(rbr)
gf.to_parquet('naipv1_1.parquet', schema_version='1.1.0')

round_trip = gpd.read_parquet('naipv1_1.parquet')
batch = stac_geoparquet.arrow.stac_table_to_items(round_trip.to_arrow())
ic = pystac.ItemCollection(batch)
[i.validate() for i in ic.items]
```
